pkg/webhook/dockerhub.go

```diff
@@ -63,34 +63,24 @@ func (s *dockerPushHook) Handle(c *gin.Context) {
 		return
 	}
 
-	// This will clone the repo before responding to the webhook. We need
-	// to make sure that this doesn't cause the hook to hang up.
-	brigadeJS, err := s.getFile(commit, "./brigade.js", proj)
-	if err != nil {
-		log.Printf("aborting DockerImagePush event due to error: %s", err)
-		c.JSON(http.StatusBadRequest, gin.H{"status": "brigadejs not found"})
-		return
-	}
-
-	go s.notifyDockerImagePush(proj, commit, body, brigadeJS)
+	go s.notifyDockerImagePush(proj, commit, body)
 	c.JSON(200, gin.H{"status": "Success"})
 }
 
-func (s *dockerPushHook) notifyDockerImagePush(proj *brigade.Project, commit string, payload, brigadeJS []byte) {
-	if err := s.doDockerImagePush(proj, commit, payload, brigadeJS); err != nil {
+func (s *dockerPushHook) notifyDockerImagePush(proj *brigade.Project, commit string, payload []byte) {
+	if err := s.doDockerImagePush(proj, commit, payload); err != nil {
 		log.Printf("failed dockerimagepush event: %s", err)
 	}
 
 }
 
-func (s *dockerPushHook) doDockerImagePush(proj *brigade.Project, commit string, payload, brigadeJS []byte) error {
+func (s *dockerPushHook) doDockerImagePush(proj *brigade.Project, commit string, payload []byte) error {
 	b := &brigade.Build{
 		ProjectID: proj.ID,
 		Type:      "image_push",
 		Provider:  "dockerhub",
 		Commit:    commit,
 		Payload:   payload,
-		Script:    brigadeJS,
```

Comment:
Can we really just defer determining the script to be executed for a build like this?

As far as I understand, brigade-controller as of today doesn't have an ability to populate missing brigade.js(=`script` in a secret representing an brigade event) - it looks like we have to populate it on gateway-side. 

I'm writing this referring the followings:

- https://github.com/Azure/brigade/search?utf8=%E2%9C%93&q=syncSecret&type=
- https://github.com/Azure/brigade/search?utf8=%E2%9C%93&q=Script&type=

Please correct me if I'm missing something!

Reply:
Just noticed that brigade does handle this case [here](https://github.com/Azure/brigade/blob/2d6281b12ed97d70ddb675dc79a5e70dc8722583/brigade-controller/cmd/brigade-controller/controller/handler.go#L137-L162).
Sorry for bothering!

For anyone interested, in nutshell brigade seems to create a vcs-sidecar(as an init-container) to populate `/vcs/brigade.js` only when `script` isn't included in an event. Then, the brigade-worker runs either `/etc/brigade/script`(corresponds to the `script` included in a `brigade.Build`=secret) or `/vcs/brigade.js` [according to this code](https://github.com/Azure/brigade/blob/2d6281b12ed97d70ddb675dc79a5e70dc8722583/brigade-worker/prestart.js). So, if we omit a `script` in a brigade.Build coming from the cr-gateway, the worker just tries to git-clone the necessary script=`brigade.js` from the vcs repo. No problem.

Reply:
Thanks for taking the time to look through all of that.